robolectric-shadows/shadows-core/src/main/resources/org/robolectric/shadows/ShadowActivity.java.vm

```diff
@@ -530,6 +531,37 @@ public class ShadowActivity extends ShadowContextThemeWrapper {
     invoker.call("onRestoreInstanceState", Bundle.class).with(outState);
     invoker.call("onResume").withNothing();
   }
+  
+  @Implementation
+  public int getChangingConfigurations() {
+    return changingConfiguration;
+  }
+  
+  @Implementation
+  public boolean isChangingConfigurations() {
+    return changingConfiguration != 0;
+  }
+  
+  public void recreateWithConfigChanges(int changedConfigBits) {
```

Comment:
I think I would like to push this code into the activity controller rather than add it to the shadow. You might also consider setting the Activity fields:-

    boolean mChangingConfigurations = false;
    int mConfigChangeFlags;

directly via reflection rather than shadowing the method entirely, not really sure its much better, but this seems to be the interface it has with ActivityThread:-

https://github.com/android/platform_frameworks_base/blob/258a554f6299749322750a286d802267c240ec31/core/java/android/app/ActivityThread.java#L3947